### ISSUE TITLE
Binary Image Gets Corrupted When PUT Over FTP 

### ISSUE THREAD

**Issue by agilliam-cmg** [mscdex/node-ftp#253]
Every image I PUT() to a server using FTP client gets corrupted.  I've tested with jpeg and png, all are damaged during transfer.  Some characters are missing and some are displaced:

`    async sendImage(image) {
        try {
            const credentials = await getFtpCredentials(this.platform);
            console.log('Sending files via FTP client');
            return new Promise((resolve, reject) => {
                let ftpClient = new ftp();
                ftpClient.on('ready', () => {
                    console.log(`Putting ${image.path} onto server`);
                    // Set transfer type to binary
                    ftpClient.binary(err => {
                        if(err)
                            reject(err);
                        else
                            // image.content is binary string data
                            ftpClient.put(image.content, image.path, (err) => {
                                if (err) {
                                    reject(err);
                                }
                                console.log('Closing FTP connection');
                                ftpClient.end();
                                resolve(`FTP PUT for ${image.path} successful!`);
                            });

                    });
                });
                console.log('Connecting to FTP server');
                ftpClient.connect(credentials);
            });
`
Here's a copy of the [image](https://www.dropbox.com/s/264w6u5yromvdpx/Snoopy_Original_Uncorrupted.png?dl=0) before sending over FTP and here's a [corrupted copy](https://www.dropbox.com/s/e8hbfg47cd17t1f/Snoopy_After_FTP_Corrupted.png?dl=0) after.  Please compare the two using a binary editor or regular notepad.  I hope this issue can be resolved as I like this library!

Thanks!

**Comment by besserwisser** [comment]
Have you found a workaround?

**Comment by agilliam-cmg** [comment]
@besserwisser No, I'm using the [basic-ftp](https://www.npmjs.com/package/basic-ftp) library instead.
